binaries/workerserver/main.go

```diff
@@ -15,26 +17,18 @@ var httpPort = flag.Int("http_port", 9091, "port to serve http on")
 
 func main() {
 	flag.Parse()
-	go serveHTTP()
+	stat, _ := stats.NewCustomStatsReceiver(stats.NewFinagleStatsRegistry, 15*time.Second)
+	stat = stat.Precision(time.Millisecond)
+	endpoints.RegisterStats("/admin/metrics.json", stat)
+	endpoints.RegisterHealthCheck("/")
+	go endpoints.Serve(fmt.Sprintf(":%d", *httpPort))
 
 	protocolFactory := thrift.NewTBinaryProtocolFactoryDefault()
 	transportFactory := thrift.NewTTransportFactory()
 
-	err := server.Serve(server.NewHandler(), fmt.Sprintf(":%d", *thriftPort), transportFactory, protocolFactory)
+	handler := server.NewHandler(stat.Scope("workerserver"))
```

Comment:
Good to know. I like having it to copy and paste the json output, but
that's not so useful when we switch to viz.

On Mon, Jul 11, 2016 at 12:32 PM, Caitie McCaffrey <notifications@github.com
> wrote:

> In binaries/workerserver/main.go
> <https://github.com/scootdev/scoot/pull/53#discussion_r70321352>:
>
> >
> >  	protocolFactory := thrift.NewTBinaryProtocolFactoryDefault()
> >  	transportFactory := thrift.NewTTransportFactory()
> >
> > -	err := server.Serve(server.NewHandler(), fmt.Sprintf(":%d", *thriftPort), transportFactory, protocolFactory)
> > +	handler := server.NewHandler(stat.Scope("workerserver"))
>
> You don't have to prefix stats with the service name, some services do
> some don't. But just fyi absorber exports (service, source(host), metric
> name) so we already have this info.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scootdev/scoot/pull/53/files/0455468671e9ac9d8a8c02d478eb32039d85d05a#r70321352>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABc5CTwYSsrRp3cS92mXvm91MiM6oKjvks5qUpphgaJpZM4JJmGF>
> .
>